src/main/java/karuna/karuna_backend/Authentication/AuthController.java

```diff
@@ -0,0 +1,109 @@
+package karuna.karuna_backend.Authentication;
+import io.jsonwebtoken.Jwt;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import jakarta.validation.Valid;
+import karuna.karuna_backend.Authentication.DTO.*;
+import karuna.karuna_backend.Authentication.JWT.JwtTokenService;
+import karuna.karuna_backend.Authentication.Utils.AuthenticationUtil;
+import karuna.karuna_backend.DTO.UserDTO;
+import karuna.karuna_backend.Errors.AuthenticateExceptions.LoginPasswordAuthenticationException;
+import karuna.karuna_backend.Errors.DTO.AuthenticationErrorResponse;
+import karuna.karuna_backend.Errors.DTO.DataIntegrityErrorResponse;
+import karuna.karuna_backend.Errors.DTO.JwtErrorResponse;
+import karuna.karuna_backend.Models.User;
+import karuna.karuna_backend.Services.UserService;
+import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+public class AuthController {
+
+    private final UserService service;
+    private final JwtTokenService jwtTokenService;
+
+    @Autowired
+    public AuthController(UserService userService,
```

Comment:
if we use lombok in project, this construtctor can be annotation

Reply:
True